src/Pages/ProductDetail.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { useParams } from 'react-router'
+import { useStore } from '../Store/context'
+import { FcLikePlaceholder, FcLike } from "react-icons/fc";
+import { FaStar } from "react-icons/fa";
+import "./ProductDetail.css"
+import axios from 'axios';
+import { Link } from 'react-router-dom';
+
+const ProductDetail = () => {
+
+    const {productId} = useParams()
+    const {state, dispatch} = useStore()
+
+    
+
+    const  SelectedProduct = state.products.find(product => product.id === productId)
+    const {id, img, name, price, desc, rating, discount, isWishlist, inCart} = SelectedProduct
+    const toggleWishlist = () => {
+        if(isWishlist){
+            dispatch({type: "REMOVE_FROM_WISHLIST", payload: id})
+        } else {
+            async function fetchData() {
```

Comment:
fetchData can be extracted out as a function and reused with params to call dispatch